[DON'T MERGE] GCS Checkpointing Testing Workload modification

bernardhan33 opened the pull request:
# This is created as a draft PR for GCS internal members to comment. This will not be merged to main.

## Checkpointing a 64B model through MaxText
- Read and Write times to be collected and sent to GCS buckets before a separate Python program aggregates and uploads to BQ. I've created b/353631904 to track the improvement of letting each pod to write directly to BQ, which is currently blocked by needed nodepool recreation.
- A sample YAML file is provided for code review purposes.

bernardhan33 committed:
check in modification for checkpointing workload

MattIrv commented on MaxText/configs/base.yml:
```diff
@@ -390,3 +390,6 @@ enable_single_controller: False
 allow_split_physical_axes: False
+
+# Attributes needed for the MaxText checkpointing loop using standalone_checkpointer.
+gcs_metrics_bucket: distributed-checkpointing-metrics
```
Should there be / is there any ability to provide a folder or path instead of just the bucket?

MattIrv committed:
[gcs-team] GCS Checkpointing benchmark feature updates (#836)

* Configurable number of parameters and save/restore loop.

This adds the following features to the gcs-checkpointing setup:
- Configurable number of parameters by setting the $PARAMETERS env
  variable
- Reloads all checkpoints immediately after saving them
- Adds a configurable minimum step time (when saving checkpoints)

* Address code review comments

* Address round 2 of code review

bernardhan33 committed:
Add node attributes; Fix GCS upload; Add checkpointID to checkpointing workload (#850)

* Use quantized kv values in einsum.

* Add support for proxy backend when running maxengine_server

* Add tp support for MoE

* Incorporate GKE's stopgap solution in the absence of in-place pod restart

* Add enable_model_warmup flag for AOT compilation at model server start

* Install Transformer Engine from github for stable and nightly mode

* Add support for cloud logging in emergency checkpoint manager.

* Adding support for creating maxtext docker images with jax-ss

* Copybara import of the project:

--
a3ed121cfbebe350ba60ab06d92327b8a3dc76b6 by RissyRan <ran.rissy@gmail.com>:

Update tp sharding annotation for MoE block

COPYBARA_INTEGRATE_REVIEW=https://github.com/google/maxtext/pull/788 from google:tp_update a3ed121cfbebe350ba60ab06d92327b8a3dc76b6
PiperOrigin-RevId: 653803904

* Llama2 forward pass checker

* support pre-tokenized dataset and fix hf issues

* Fix a Goodput upload to TB bug

* Temp fix for c4_mlperf pipeline

* Adds new remat policy to save out_proj activations during training.

PiperOrigin-RevId: 655745941

* Add wrap argument when creating XAOT topology mesh

* Refactor code

* Fix maxtext profiler failures

* Changes to Copybara to include renamed files

PiperOrigin-RevId: 656444119

* Copybara import of the project:

--
adea1b1e8e4006de77f09e3cf09ba661a929bb93 by Param Bole <parambole@google.com>:

Refactoring Maxtext-JAX-Stable-Stack Build Process

COPYBARA_INTEGRATE_REVIEW=https://github.com/google/maxtext/pull/796 from google:pb_jax_ss_maxtext adea1b1e8e4006de77f09e3cf09ba661a929bb93
PiperOrigin-RevId: 656474261

* Refactoring Maxtext-JAX-Stable-Stack Build Process

* Remove pre-commit install from setup.sh

* Internal change

PiperOrigin-RevId: 657990080

* Pin jax stable version > 0.4

* Revert internal change to maxengine

* Add unique_indices to jnp.take

* Removing unused arg PLATFORM

* Add Gemma2 9B Config, Window Attention and attention logic soft cap.

Fixing attention_type type annotation

Fixing trailing-whitespace

Updating 9B test_gemma.sh scripts

Update gemma 9B 2_test_gemma.sh

Updating gemma 9B checkpoint directory

Update default BASE_OUTPUT_PATH in gemma 9B

Remove finetunning script for now as it OOMs on TPU v4

Updating gemma-9B to gemma2-9B

Fixing comment in gemma2-9b

Updating comments on attention kernel and attention_type.

* Removing unused agument PLATFORM from preflight.sh

* Revert "Add unique_indices to jnp.take"

* Consolidate usages of `MultiprocessingOptions` and `AsyncOptions`. Formalize `StandardCheckpointer` as an `AsyncCheckpointer` that doesn't require a `CheckpointArgs` object, and instead allows directly passing the state and extra args.

PiperOrigin-RevId: 660588942

* Remove unused checkpoint import

* Add A3 AoT compilation support using mock GPU client

Support XAOT on GPU

Update mock GPU client API and avoid initializing backend in pyconfig

Allow NoneType in SystemCharacteristics

Fix spacing

Use Optional instead of Union

Make topology_name optional and improve readme

Use dot_product attention in README example

* generates padding batch in hf pipeline to improve data utilization

* Add Gemma2 Support to MaxText: Gemma2 Decoder Layers, Checkpoint Converter, Config Files, Flop Calculation and Run Scripts

* Change scripts for llama2 7b a3 GPU to use llama2_7b_gpu.yml

PiperOrigin-RevId: 662216695

* Copybara import of the project:

--
80a49ff7a7239b09433d49c7501c4c558727bc30 by RissyRan <ran.rissy@gmail.com>:

Add instruction for Mixtral

COPYBARA_INTEGRATE_REVIEW=https://github.com/google/maxtext/pull/822 from google:mixtral_instruct 80a49ff7a7239b09433d49c7501c4c558727bc30
PiperOrigin-RevId: 662219295

* [MLPerf][GPT3] Bypass setting eval_interval in using synthetic dataset

* Removing the resgistration of the proxy backend used by Pathways.

If needed, this is handled by a separate, Pathways specific utilities package.

* Support AoT in 16-vm GPU Llama2 train script

* Initial grad accumulate with scan

Gradient accumulation

manually tested

globals are sus

Add grad accumulation test

Gradient accumulation config assert

Gradient accumulation config assert

pylint

fixed tests

global and micro batch size

Clean up with microbatches

Gradient accumulation

Gradient accumulation

Gradient accumulation

Gradient accumulation

Gradient accumulation

Gradient accumulation

Gradient accumulation

grad acc

grad acc

* Add support for local sliding window attention in TPU splash_attention

* add kl divergence and more logging for forward_pass_logit_checker

* Add dropping strategy

* add node attributes; fix GCS upload; add checkpointID

* add header to the checkpointing workload

* add rank to node attributes list

---------

Co-authored-by: Mitali Singh <msingh@google.com>
Co-authored-by: Shaurya Gupta <shaurya.gup@gmail.com>
Co-authored-by: maxtext authors <no-reply@google.com>
Co-authored-by: RissyRan <ran.rissy@gmail.com>
Co-authored-by: Xuefeng Gu <xfgu@google.com>
Co-authored-by: Vivian Wu <vivianrwu@google.com>
Co-authored-by: michelle-yooh <yooh@google.com>
Co-authored-by: Abhinav Singh <abhinavsing@google.com>
Co-authored-by: Param Bole <parambole@google.com>
Co-authored-by: khatwanimohit <mohitkhatwani@google.com>
Co-authored-by: aireenmei <aireenmei@gmail.com>
Co-authored-by: Dipannita Shaw <dishaw@google.com>
Co-authored-by: ZhiyuLi-goog <lizhiyu@google.com>
Co-authored-by: Raymond Zou <raymondzou@google.com>
Co-authored-by: hengtaoguo <hengtaoguo@google.com>
Co-authored-by: Gagik Amirkhanyan <agagik@google.com>
Co-authored-by: Colin Gaffney <cpgaffney@google.com>
Co-authored-by: gobbleturk <mattdavidow@google.com>
Co-authored-by: Jon Bolin <jonbolin@google.com>
Co-authored-by: Zhaoyue Cheng <joychengzhaoyue@gmail.com>
Co-authored-by: Luke Baumann <lukebaumann@google.com>

bernardhan33 committed:
Check in changes to support older checkpoint deletion and customized checkpoint sizes (#882)

mkmg committed:
Adding 405b config

mkmg committed:
Adding 512b config

bernardhan33 committed:
merge with main and update

bernardhan33 committed:
resolve more conflicts from main

MattIrv committed:
Update gcs-checkpointing branch with latest changes from main (#1198)

* [Inference Perf] Add autotuned xla flags to improve latency for v6e

Reason:
Use auto tune to find a better set of xla flag settings that
can improve inference latency. It gives ~10% latency reduction
for generate step and doesn't hurt prefill.

Microbenchmark result
BEFORE:
Prefill benchmark results for length 1024:
Per prefill step per device:
Prefill step average time: 60.695 ms <<<<<<<<<<<<<<<<<<<<<<<<<<
Prefill and insert benchmark results for length 1024:
Prefill + Insert step average time: 63.350 ms
AutoRegressive results:
AR step average time: 63.547 ms <<<<<<<<<<<<<<<<<<<<<<<<<<<<<

AFTER:
Prefill benchmark results for length 1024:
Per prefill step per device:
Prefill step average time: 60.451 ms<<<<<<<<<<<<<<<<<<<<<<<<<<<< 0.5% better
Prefill and insert benchmark results for length 1024:
Prefill + Insert step average time: 62.944 ms
AutoRegressive results:
AR step average time: 57.327 ms<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< 9.8% better

* adding new JSS builds for verification

* Adding new JSS builds for verification (#1038)

* Add missing period.

* Add config param to enable monitoring Goodput with Pathways enabled

* Disable checkpoint cloud logger in smoke tests.

PiperOrigin-RevId: 697790035

* Enable orbax checkpoint cloud logger as default.

* Updating Jax Stable Stack BaseImage path

* Scan and remat only pipeline iterations

* Add custom 4x64 mesh for single slice

* Adding support for mixed precision drq.

* Set Orbax logger to False.

* Adds 405B script for GPU for consistency.

* Optimizer memory offload

* Add 32x8 custom split

* Set block_q_dq and block_kv_dq to None if use_fused_bwd_kernel is enabled

* More flexible offline benchmarking.

Checks for existing tokenizer and checkpoint.

.

* llama3 405b and Disaggregation prototyping

Add llama3 405b 70b inference experimental support on v5e-16 and fetch the KV cache sharding from model forward in decode phase and use different sharding for prefill and decode.

correct model name for llama-2 70b

add a flag to replicate the quantization scale to avoid XLA inefficient fusion for 2d sharding

Disaggregation prototyping

avoid redundant decode state initialization

disagg code change for v6e-8

rename replicate_qua_scale as replicate_quant_scale

* Parallelizes accuracy eval.

* This PR adds a template for PR descriptions, aimed to provide sufficient details
about the changes being made.

* Remove jon as maxtext owner (sad)

* Fix MoE related tests

* Add checkpoint topology discovery for the Replicator Service

* Fix local restore by re-mapping device ids directly instead of inferring them from how process indexes changed across restarts with some false assumptions.

PiperOrigin-RevId: 700737164

* Compact the number of variables for the prefill result cache to reduce Python layer overhead

* add more MoE tests

* update setup_gcsfuse for better perf

* Assert multiple slices available when requesting DCN parallelisms

* disable tf GPU visibility

* Add llama 3.1 70b config

* added comment

* Update replicator.yaml to include framework and num_slices information

* Support JAX_VERSION for nightly mode on GPU

* Added a custom_wheel mode for building the dependency image.

This mode is the same as nightly except that after nightly is installed, any file in `maxtext/*.whl` is forcefully reinstalled.

* clean up pipeline config setting in its own method

* Fixes for dropping

* Fix setup_training_state

* Change awk regex command to capture the coordinate address properly

* point to new jax github location in documentation

PiperOrigin-RevId: 703063210

* added Sliding Window Attention support for cudnn flash attn

* Fixes non-hashable error in ragged attn.

* Add new remat policy for save_dot_except_mlp with context

* Allows env var for logging level.

Uses env var for logging control

Fixes cmd line args processing.

Temp change for testing accuracy.

Uses MaxText tokenizer.

Removes test flag

* make sharding_tolerance configurable

* Adding a .dockerignore file

This will ignore the .git directory when copying the maxtext directory
into the docker images

* Modify axis orders for optimal PP

* DPO running

* initial jdi

* Fix bug to suport non-2x max_target_length

* fix to TF without GPU throwing errors

* Apply quantization on megablox kernel; support both training and serving

* Use jax.lax.dot_general when both lhs and rhs are not quantized

* Fix lib version mismatch between orbax and grain

* change the mask type default to None for causal

* Add Llama 3.1 with 8192 seq len Trillium config

* Add ran as maxtext reviewer

* Add options to reduce manual work in e2e JetStream benchmark tests.

* Recommend using v5e instead of v4

* Update First_run.md

* Fix typo in custom remat config comments

* Update type casting for activation

* Better save message for async checkpoint saving

* Print memory stats after start_step

Update the label for print_mem_stats during params intialization

Remove accidentally added xpk

* Bumps up Transformer Engine to 1.13 from 1.5.

* Fix pylint

* Remove the custome te builder file.

* Add Dockerfile and entrypoint script for the maxengine-server image

* Add smaler v5e topologies to train_compile

* Pylint for forks, some condition as unit tests

* [MoE] notebook for numerical verification

* Move tests from UnitTests.yml to tests files.

This change moves out train and decode tests into separate files,
so they can be tested using pytest rather than explicitly being
invoked from UnitTests.yml. This will allow us, in a subsequent
PR, to parallelize the execution of these tests by utilizing
pytest parallelization support.

Several end-to-end tests are still being invoked from UnitTests.yml.
This does not seem like the right place, and will hopefully be
addressed in the future.

* Add back gpt3 test

* Update link to unit tests in Readme

* Improve build caching for GPU build.

* added support for sequence sharding with tensor parallelism

* Remove memory stat spam

* copy assets separately

* Add option to upload HLO

* Add option to skip initializing the jax distributed system

* added activation_norm_length to all models and completed performance tests

* linter fixes

* clean for linter

* added tensor_sequence support for models using pp

* linter fixes

* [MoE][Inference] Router Sharding optimization

* Update llama2_7b_4096.sh

* Fix bug in processing mixed precision quantization config

* Update max target length check

* Fixed a bug so that config parallelism is not modified during a run.

It was being modified in-place after a recent refactor.

* udpate orbax-checkpoint pin to latest

* Add option to profile periodically

* Run tests against JAX StS

* Refactor: Changing JStS tags from pinned to latest

* Add memory stats from JAX and compiled memory analysis.

* Print Jax stats only once after all iterations; add the wrap for the compile() to avoid compile OOM

* remove redundant memory_stats

* Add mixtral-8x7B offline inference script

* Incorporate Orbax emergency replicator checkpoint manager

* Incorporate Orbax emergency replicator checkpoint manager

* Incorporate Orbax emergency replicator checkpoint manager

* Incorporate Orbax emergency replicator checkpoint manager

* fix orbax to hf converter

* Bypass printing memstats when not available on pathways

* Recover, rename and move the test_profiler.py to end_to_end

restore changes

* add environment variable to change jetstream version

* Add Pathways Support to Benchmark Runner

* Add MoE block unit tests

* Add more 8x22b pre-training and decoding tests

* Copybara import of the project:

--
52882e57dd2e51205c35e1f4441744f86b44bd6e by Mohit Khatwani <mohitkhatwani@google.com>:

install torch[cpu] in test script

COPYBARA_INTEGRATE_REVIEW=https://github.com/AI-Hypercomputer/maxtext/pull/1165 from AI-Hypercomputer:mohit/fix_orbax_hf 52882e57dd2e51205c35e1f4441744f86b44bd6e
PiperOrigin-RevId: 715183387

* Refactor: updating gpu image tag

* add prefill packing for offline inference

* Integrates jax.lax.ragged_dot into MoeBlock; enabled via sparse_matmul=True and megablox=False config options. By default ragged_dot is disabled.

PiperOrigin-RevId: 715959403

* Ignore internal ragged dot smoke test

* jit entire prefill and insert for packed sequences

* Use an async checkpointer to initialize JDI in tests

* Remove usages of orbax_utils.save_args_from_target, as this function does nothing (it used to control a checkpointing behavior that has since been optimized away).

PiperOrigin-RevId: 716314210

* improve xpk_runner

* add configs

* Remove no-op flag

PiperOrigin-RevId: 716608419

* Add XLA flags to optimize PP

* Copybara import of the project:

--
f7d7e769d61a111a53711b7e84c9eb738fa1f62e by A9isha <mazumdera@google.com>:

update forward_pass_logit_checker for new dimension

COPYBARA_INTEGRATE_REVIEW=https://github.com/AI-Hypercomputer/maxtext/pull/1175 from AI-Hypercomputer:anisha-mistral-fixit f7d7e769d61a111a53711b7e84c9eb738fa1f62e
PiperOrigin-RevId: 716734551

* Update moe dropped config

* Create Getting_Started_Benchmarking.md

* remove cast_to_bf16

* Remove moe lb loss from evaluation

* enable concatenation of unpadded prompts

* Revert "Merge pull request #1184 from zhenying-liu:opt-state"

This reverts commit 8706c206f30bb77d62fc97d4c991e57cc74842cc, reversing
changes made to 514be4862119f4d3aa73db1b9261bd97bd8c2316.

* add GCP workload observability feature

* Typo to fix

* Update v5p-32b flags with SC

* fix a bug in sorting data

* Update checkpointing test for async

---------

Co-authored-by: maxtext authors <no-reply@google.com>
Co-authored-by: Wangyuan Zhang <wyzhang@google.com>
Co-authored-by: Param Bole <parambole@google.com>
Co-authored-by: gobbleturk <mattdavidow@google.com>
Co-authored-by: Dipannita Shaw <dishaw@google.com>
Co-authored-by: Abhinav Singh <abhinavsing@google.com>
Co-authored-by: Raymond Zou <raymondzou@google.com>
Co-authored-by: Mitali Singh <msingh@google.com>
Co-authored-by: Lance Wang <lancewang@google.com>
Co-authored-by: Mohit Khatwani <mohitkhatwani@google.com>
Co-authored-by: Rissy Ran <ran.rissy@gmail.com>
Co-authored-by: Pate Motter <patemotter@google.com>
Co-authored-by: Zhihao Shan <zhihaoshan@google.com>
Co-authored-by: Alex Shraer <shralex@gmail.com>
Co-authored-by: Xuefeng Gu <xfgu@google.com>
Co-authored-by: Colin Gaffney <cpgaffney@google.com>
Co-authored-by: aireenmei <aireenmei@gmail.com>
Co-authored-by: Keshav Balasubramanian <keshavb@nvidia.com>
Co-authored-by: Branden Vandermoon <bvandermoon@google.com>
Co-authored-by: Luke Baumann <lukebaumann@google.com>
Co-authored-by: Vijaya <mailvijayasingh@gmail.com>
Co-authored-by: michelle-yooh <yooh@google.com>
Co-authored-by: Jake Harmon <jakeharmon@google.com>
Co-authored-by: Md Fahim Faysal Khan <mdfahimfaysa@nvidia.com>
Co-authored-by: Doris26 <zouyujun526@gmail.com>
Co-authored-by: Luke Baumann <lukebaumann@users.noreply.github.com>
Co-authored-by: Robert Dyro <rdyro@google.com>
Co-authored-by: Vipan Nalla <vipannalla@google.com>
Co-authored-by: Wonpyo Park <wppark.pio@gmail.com>
Co-authored-by: ZhiyuLi-goog <lizhiyu@google.com>
Co-authored-by: Fenghui Zhang <fhzhang@google.com>
Co-authored-by: tornikeo <56438982+tornikeo@users.noreply.github.com>
Co-authored-by: Hengtao Guo <hengtaoguo@google.com>
Co-authored-by: Vivian Wu <vivianrwu@google.com>
Co-authored-by: Abhinav Goel <abgoel@nvidia.com>
Co-authored-by: wenxindongwork <wenxindong@google.com>
Co-authored-by: Jane Liu <janeliu@nvidia.com>
Co-authored-by: Sujeeth Jinesh <sujinesh@google.com>
Co-authored-by: Xiang Si <sixiang@google.com>
Co-authored-by: Obliviour <vbarr@google.com>
Co-authored-by: Benjamin Chetioui <bchetioui@google.com>
Co-authored-by: A9isha <mazumdera@google.com>
Co-authored-by: Johnny Yang <johnnyyang@google.com>
Co-authored-by: Anfal Siddiqui <anfals@google.com>

jdnurme committed:
Update gcs-checkpointing with phase_2 ramdisk changes from main (#1276)

* make sharding_tolerance configurable

* Adding a .dockerignore file

This will ignore the .git directory when copying the maxtext directory
into the docker images

* Modify axis orders for optimal PP

* DPO running

* initial jdi

* Fix bug to suport non-2x max_target_length

* fix to TF without GPU throwing errors

* Apply quantization on megablox kernel; support both training and serving

* Use jax.lax.dot_general when both lhs and rhs are not quantized

* Fix lib version mismatch between orbax and grain

* change the mask type default to None for causal

* Add Llama 3.1 with 8192 seq len Trillium config

* Add ran as maxtext reviewer

* Add options to reduce manual work in e2e JetStream benchmark tests.

* Recommend using v5e instead of v4

* Update First_run.md

* Fix typo in custom remat config comments

* Update type casting for activation

* Better save message for async checkpoint saving

* Print memory stats after start_step

Update the label for print_mem_stats during params intialization

Remove accidentally added xpk

* Bumps up Transformer Engine to 1.13 from 1.5.

* Fix pylint

* Remove the custome te builder file.

* Add Dockerfile and entrypoint script for the maxengine-server image

* Add smaler v5e topologies to train_compile

* Pylint for forks, some condition as unit tests

* [MoE] notebook for numerical verification

* Move tests from UnitTests.yml to tests files.

This change moves out train and decode tests into separate files,
so they can be tested using pytest rather than explicitly being
invoked from UnitTests.yml. This will allow us, in a subsequent
PR, to parallelize the execution of these tests by utilizing
pytest parallelization support.

Several end-to-end tests are still being invoked from UnitTests.yml.
This does not seem like the right place, and will hopefully be
addressed in the future.

* Add back gpt3 test

* Update link to unit tests in Readme

* Improve build caching for GPU build.

* added support for sequence sharding with tensor parallelism

* Remove memory stat spam

* copy assets separately

* Add option to upload HLO

* Add option to skip initializing the jax distributed system

* added activation_norm_length to all models and completed performance tests

* linter fixes

* clean for linter

* added tensor_sequence support for models using pp

* linter fixes

* [MoE][Inference] Router Sharding optimization

* Update llama2_7b_4096.sh

* Fix bug in processing mixed precision quantization config

* Update max target length check

* Fixed a bug so that config parallelism is not modified during a run.

It was being modified in-place after a recent refactor.

* udpate orbax-checkpoint pin to latest

* Add option to profile periodically

* Run tests against JAX StS

* Refactor: Changing JStS tags from pinned to latest

* Add memory stats from JAX and compiled memory analysis.

* Print Jax stats only once after all iterations; add the wrap for the compile() to avoid compile OOM

* remove redundant memory_stats

* Add mixtral-8x7B offline inference script

* Incorporate Orbax emergency replicator checkpoint manager

* Incorporate Orbax emergency replicator checkpoint manager

* Incorporate Orbax emergency replicator checkpoint manager

* Incorporate Orbax emergency replicator checkpoint manager

* fix orbax to hf converter

* Bypass printing memstats when not available on pathways

* Recover, rename and move the test_profiler.py to end_to_end

restore changes

* add environment variable to change jetstream version

* Add Pathways Support to Benchmark Runner

* Add MoE block unit tests

* Add more 8x22b pre-training and decoding tests

* Copybara import of the project:

--
52882e57dd2e51205c35e1f4441744f86b44bd6e by Mohit Khatwani <mohitkhatwani@google.com>:

install torch[cpu] in test script

COPYBARA_INTEGRATE_REVIEW=https://github.com/AI-Hypercomputer/maxtext/pull/1165 from AI-Hypercomputer:mohit/fix_orbax_hf 52882e57dd2e51205c35e1f4441744f86b44bd6e
PiperOrigin-RevId: 715183387

* Refactor: updating gpu image tag

* add prefill packing for offline inference

* Integrates jax.lax.ragged_dot into MoeBlock; enabled via sparse_matmul=True and megablox=False config options. By default ragged_dot is disabled.

PiperOrigin-RevId: 715959403

* Ignore internal ragged dot smoke test

* jit entire prefill and insert for packed sequences

* Use an async checkpointer to initialize JDI in tests

* Remove usages of orbax_utils.save_args_from_target, as this function does nothing (it used to control a checkpointing behavior that has since been optimized away).

PiperOrigin-RevId: 716314210

* improve xpk_runner

* add configs

* Remove no-op flag

PiperOrigin-RevId: 716608419

* Add XLA flags to optimize PP

* Copybara import of the project:

--
f7d7e769d61a111a53711b7e84c9eb738fa1f62e by A9isha <mazumdera@google.com>:

update forward_pass_logit_checker for new dimension

COPYBARA_INTEGRATE_REVIEW=https://github.com/AI-Hypercomputer/maxtext/pull/1175 from AI-Hypercomputer:anisha-mistral-fixit f7d7e769d61a111a53711b7e84c9eb738fa1f62e
PiperOrigin-RevId: 716734551

* Update moe dropped config

* Create Getting_Started_Benchmarking.md

* remove cast_to_bf16

* Remove moe lb loss from evaluation

* enable concatenation of unpadded prompts

* Revert "Merge pull request #1184 from zhenying-liu:opt-state"

This reverts commit 8706c206f30bb77d62fc97d4c991e57cc74842cc, reversing
changes made to 514be4862119f4d3aa73db1b9261bd97bd8c2316.

* add GCP workload observability feature

* Typo to fix

* Update v5p-32b flags with SC

* fix a bug in sorting data

* Update checkpointing test for async

* Add MoE AoT tests

* Internal Change

PiperOrigin-RevId: 719554528

* Reverts 8dd66612ed53a9b0c5c7c3b014d4e75a956d3853

PiperOrigin-RevId: 720275184

* Add on-device runner and pythonic v5e model configs

* fixing types and updating main in maxengine and inference_microbenchmarks.

* MMLU Benchmark

* Add a base image for GPU build.

* Notify on MoxText build failures.

The action will find the latest open issue with the label "build failed" and add a comment.
If no such issue is open, it will instead open a new issue. We exclude failures on pull requests
because they represent in-development work where failures are more expected.

Reference: https://github.com/jayqi/failed-build-issue-action

* make synthetic data more realistic

* Updated XLA Flags for SparseCore

* Skip falky tests in hf_data_processing_test.py

* Small fix to build failure notification flow.

* Merge

Remote Python Support for Benchmark Runner

Add Remote Python Recipe

Clean up

Add Delete Capabilities

Minor edits

Fix

Fix problems

Changes

Changes

Fixes

Undo Weird Merge

Save work

Break out args helper

Fix file imports

Move around stuff

cleanup

Fix

Clean up comment

whitespace

Fix issues

Fix lint

* Make run_name optional for on-device benchmark runner

* Maintain fp32 for optimizer state when offloading is enabled

* Update Run_Gemma.md with correct links.

* make attention fp32 flags apply during inference

* move float32_qk_product and float32_logits to config

* Fixes max threshold in RA tests.

* address review comments

* Add an option to disable tensorboard

We have found that tensorboard does not support
our network based storage backend. Add an option to disable it.

* PP: All gather weights once at start

* Add a basic unit test for prefill in maxengine

* Add GPU MoE e2e test script for XLML testing

* Pin latest goodput package

* Add Support for FP8 Quantization in Pipeline Parallelism

* adds metrics export to ragged dot smoke test

PiperOrigin-RevId: 723218740

* Optimize the device assignment for TPU v6e

benchmark results:

before:
================================================
MLPerf Results Summary
================================================
SUT name : PySUT
Scenario : Offline
Mode     : PerformanceOnly
Samples per second: 10.3444
Tokens per second: 2931.8419
Result is : VALID
  Min duration satisfied : Yes
  Min queries satisfied : Yes
  Early stopping satisfied: Yes

after and with flag enabled:
================================================
MLPerf Results Summary
================================================
SUT name : PySUT
Scenario : Offline
Mode     : PerformanceOnly
Samples per second: 11.5125
Tokens per second: 3272.4449
Result is : VALID
  Min duration satisfied : Yes
  Min queries satisfied : Yes
  Early stopping satisfied: Yes

* Fix failure notification action.

* Update README.md to remove preview tag.

* Ahead of time compilation with auto layouts for parameters and decode.

* Fix indentation in RunTests

* Get partition spec only when needed

* JStS for GPUs Preview Integration

* squash commit (#1195)

* Notify build failure action should always succeed

* Copybara import of the project:

--
052c76f996b006408ea0b0c7c2985eb6f1f885bc by Rissy Ran <ran.rissy@gmail.com>:

Onboard DeepSeek MoE with shared experts

COPYBARA_INTEGRATE_REVIEW=https://github.com/AI-Hypercomputer/maxtext/pull/1259 from AI-Hypercomputer:ds_v3 052c76f996b006408ea0b0c7c2985eb6f1f885bc
PiperOrigin-RevId: 725679454

* Yarn Rotary Embedding for DeepSeek v3

Change deepseek test to be manual because of Torch dependancy.

rename test to check_mla_vs_reference.py

* updating trillium benchmarking scripts and README

* Run Maxtext unit test on candidate JStS Image

* Add 8x22b config to benchmark

* Integrate MaxText with Goodput v5

* Enable restore with replictor emergency checkpoint manager

---------

Co-authored-by: Doris26 <zouyujun526@gmail.com>
Co-authored-by: Luke Baumann <lukebaumann@google.com>
Co-authored-by: maxtext authors <no-reply@google.com>
Co-authored-by: Luke Baumann <lukebaumann@users.noreply.github.com>
Co-authored-by: gobbleturk <mattdavidow@google.com>
Co-authored-by: Robert Dyro <rdyro@google.com>
Co-authored-by: Vipan Nalla <vipannalla@google.com>
Co-authored-by: Wonpyo Park <wppark.pio@gmail.com>
Co-authored-by: Md Fahim Faysal Khan <mdfahimfaysa@nvidia.com>
Co-authored-by: ZhiyuLi-goog <lizhiyu@google.com>
Co-authored-by: Raymond Zou <raymondzou@google.com>
Co-authored-by: Fenghui Zhang <fhzhang@google.com>
Co-authored-by: tornikeo <56438982+tornikeo@users.noreply.github.com>
Co-authored-by: RissyRan <ran.rissy@gmail.com>
Co-authored-by: Hengtao Guo <hengtaoguo@google.com>
Co-authored-by: Lance Wang <lancewang@google.com>
Co-authored-by: Vivian Wu <vivianrwu@google.com>
Co-authored-by: Alex Shraer <shralex@gmail.com>
Co-authored-by: Abhinav Goel <abgoel@nvidia.com>
Co-authored-by: Mohit Khatwani <mohitkhatwani@google.com>
Co-authored-by: wenxindongwork <wenxindong@google.com>
Co-authored-by: Param Bole <parambole@google.com>
Co-authored-by: Jane Liu <janeliu@nvidia.com>
Co-authored-by: Xuefeng Gu <xfgu@google.com>
Co-authored-by: Sujeeth Jinesh <sujinesh@google.com>
Co-authored-by: Xiang Si <sixiang@google.com>
Co-authored-by: Colin Gaffney <cpgaffney@google.com>
Co-authored-by: Obliviour <vbarr@google.com>
Co-authored-by: Benjamin Chetioui <bchetioui@google.com>
Co-authored-by: A9isha <mazumdera@google.com>
Co-authored-by: Johnny Yang <johnnyyang@google.com>
Co-authored-by: Anfal Siddiqui <anfals@google.com>
Co-authored-by: Mitali Singh <msingh@google.com>
Co-authored-by: Gagik Amirkhanyan <agagik@google.com>
Co-authored-by: yangyuwei <yangyuwei@google.com>
Co-authored-by: yangyuwei <143511705+yangyuwei@users.noreply.github.com>
Co-authored-by: Philip Monk <169196560+philip-essential@users.noreply.github.com>
Co-authored-by: Roshani Narasimhan <roshanin@google.com>
Co-authored-by: Pate Motter <patemotter@google.com>
Co-authored-by: pavanholla <pavanholla@gmail.com>
Co-authored-by: liurupeng <rupliu@google.com>
Co-authored-by: michelle-yooh <yooh@google.com>
Co-authored-by: Dipannita Shaw <shaw.dipannita@gmail.com>
Co-authored-by: Hao Luo <haoluo@google.com>
Co-authored-by: Sai Balakrishnan <saisanthosh@google.com>
Co-authored-by: Brendan Slabe <slabe@google.com>
